### PR TITLE
chore: create a delta index to cover the unindexed fragments

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/lance-index/src/optimize.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub struct MergeIndex {}
+
+pub enum OptimizeAction {
+    Append,
```
Comment (chebbyChefNEQ) [inline rust/lance-index/src/optimize.rs:18]: should we just make `CreateDelta` a mode of `Append`? say `Append(create_new_fragment)`

Comment (eddyxu) [reply rust/lance-index/src/optimize.rs:18]: Lemme think about it.  The intent was that one would not merge with the previous index, and the other would merge with the previous index.

Comment (eddyxu) [reply rust/lance-index/src/optimize.rs:18]: Probally we just have one option to determine how many previous index to be merged together?

**Comment by eddyxu** [comment]
Close in favor of #1841

**Pull request closed without merging**